File: pulsar-core/pulsar-tools/pulsar-browser/src/main/java/com/github/kklisura/cdt/protocol/v2023/events/page/FileChooserOpened.java
package com.github.kklisura.cdt.protocol.v2023.events.page;

/*-
 * #%L
 * cdt-java-client
 * %%
 * Copyright (C) 2018 - 2023 Kenan Klisura
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import com.github.kklisura.cdt.protocol.v2023.support.annotations.Experimental;
import com.github.kklisura.cdt.protocol.v2023.support.annotations.Optional;

/** Emitted only when `page.interceptFileChooser` is enabled. */
public class FileChooserOpened {

  @Experimental
  private String frameId;

  private FileChooserOpenedMode mode;

  @Experimental @Optional
  private Integer backendNodeId;

  /** Id of the frame containing input node. */
  public String getFrameId() {
    return frameId;
  }

  /** Id of the frame containing input node. */
  public void setFrameId(String frameId) {
    this.frameId = frameId;
  }

  /** Input mode. */
  public FileChooserOpenedMode getMode() {
    return mode;
  }

  /** Input mode. */
  public void setMode(FileChooserOpenedMode mode) {
    this.mode = mode;
  }

  /** Input node id. Only present for file choosers opened via an `<input type="file">` element. */
  public Integer getBackendNodeId() {
    return backendNodeId;
  }

  /** Input node id. Only present for file choosers opened via an `<input type="file">` element. */
  public void setBackendNodeId(Integer backendNodeId) {
    this.backendNodeId = backendNodeId;
  }
}

File: pulsar-core/pulsar-tools/pulsar-browser/src/main/kotlin/ai/platon/cdt/kt/protocol/events/page/FileChooserOpened.kt
@file:Suppress("unused")
package ai.platon.cdt.kt.protocol.events.page

import ai.platon.cdt.kt.protocol.support.annotations.Experimental
import ai.platon.cdt.kt.protocol.types.page.FileChooserOpenedMode
import com.fasterxml.jackson.`annotation`.JsonProperty
import kotlin.Int
import kotlin.String

/**
 * Emitted only when `page.interceptFileChooser` is enabled.
 */
data class FileChooserOpened(
  @param:JsonProperty("frameId")
  @param:Experimental
  val frameId: String,
  @param:JsonProperty("backendNodeId")
  @param:Experimental
  val backendNodeId: Int,
  @param:JsonProperty("mode")
  val mode: FileChooserOpenedMode,
)
